spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/StreamListenerAnnotationBeanPostProcessor.java

```diff
@@ -109,89 +113,251 @@ public Object postProcessAfterInitialization(final Object bean, String beanName)
 			public void doWith(final Method method) throws IllegalArgumentException, IllegalAccessException {
 				StreamListener streamListener = AnnotationUtils.findAnnotation(method, StreamListener.class);
 				if (streamListener != null) {
+					Assert.isTrue(method.getAnnotation(Input.class) == null,
+							"A @StreamListener may never be annotated with @Input. If it should listen to a specific input, " +
+									"use the value of @StreamListener instead.");
+					String methodAnnotatedInboundName = getInboundElementNameFromMethodAnnotation(streamListener);
+					String methodAnnotatedOutboundName = getOutboundElementNameFromMethodAnnotation(method);
```

Comment:
seems like the method being called here should be named: `getOutboundElementNameFromMethod(..)` since it takes the `Method` instance and *not* an annotation